indico/modules/rb/client/js/modules/admin/MapAreasPage.jsx

```diff
@@ -0,0 +1,372 @@
+        });
+
+        deleteMapAreas(deletedAreas);
+    };
+
+    drawAreas = () => {
+        if (this.map === null) {
+            return null;
+        }
+
+        const {areas} = this.props;
+        return areas.map((area) => {
+            const southWest = [area.top_left_latitude, area.top_left_longitude];
+            const northEast = [area.bottom_right_latitude, area.bottom_right_longitude];
+            const latLngBounds = new Leaflet.LatLngBounds([southWest, northEast]);
+            const center = this.map.leafletElement.latLngToContainerPoint(latLngBounds.getCenter());
+            const bottomRight = this.map.leafletElement.latLngToContainerPoint(latLngBounds.getSouthEast());
+            const onAreaClick = () => {
+                const {suppressClick} = this.state;
+                if (suppressClick) {
+                    return;
+                }
+
+                this.setState({
+                    areaProps: {
+                        bounds: {
+                            southWest: latLngBounds.getSouthWest(),
+                            northEast: latLngBounds.getNorthEast()
+                        },
+                        id: area.id,
+                        isDefault: area.is_default,
+                        name: area.name
+                    }
+                });
+            };
+
+            return (
+                <Rectangle key={`${area.name}-${center.x}-${center.y}`}
+                           bounds={[southWest, northEast]}
+                           name={area.name}
+                           isDefault={area.is_default}
+                           onClick={onAreaClick}
+                           id={area.id}>
+                    <Tooltip offset={[bottomRight.x - center.x, bottomRight.y - center.y]}
+                             direction="bottom"
+                             sticky
+                             permanent
+                             interactive>
+                        <>
+                            {area.name}
+                            {area.is_default && (
+                                <span className="area-default">★</span>
+                            )}
+                        </>
+                    </Tooltip>
+                </Rectangle>
+            );
+        });
+    };
+
+    onSubmit = (data) => {
+        const {actions: {createMapArea, updateMapAreas}} = this.props;
+        const {areaProps: {id}} = this.state;
+        let result;
+
+        if (!id) {
+            result = createMapArea(data);
+        } else {
+            result = updateMapAreas([{...data, id}]);
+        }
+
+        result.then(() => {
+            this.setState({areaProps: null});
+        });
+    };
+
+    renderModalForm = () => {
+        const {areaProps} = this.state;
+        if (!areaProps) {
+            return null;
+        }
+
+        const {id, bounds, name, isDefault} = areaProps;
+        return (
+            <FinalForm onSubmit={this.onSubmit}
```

Comment:
specify the subscription

Reply:
Oh yeah, I was actually to do that! Thanks for reminding me about it ;)

Reply:
Hm, now when I am using `<Final...` fields I started wondering whether I even have to specify subscription on the form.

Reply:
I guess an empty object would be fine here?

Reply:
If you don't use any fprops (except the submit callback) then it can indeed be empty